features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/FlagsClient.kt

```diff
@@ -0,0 +1,444 @@
+            val featureCore = sdkCore as FeatureSdkCore
+            val logger = featureCore.internalLogger
+
+            val flagsFeature = featureCore.getFeature(FLAGS_FEATURE_NAME)?.unwrap<FlagsFeature>()
+
+            if (flagsFeature == null) {
+                logger.log(
+                    InternalLogger.Level.ERROR,
+                    InternalLogger.Target.USER,
+                    { "Flags feature is not enabled. Returning NoOpFlagsClient." }
+                )
+
+                // No feature available to log through, so we use the logger directly.
+                return NoOpFlagsClient(
+                    name = name,
+                    reason = "Flags feature not enabled",
+                    logWithPolicy = { message, level ->
+                        logger.log(level, InternalLogger.Target.USER, { message })
+                    }
+                )
+            }
+            return flagsFeature.getClient(name) ?: handleClientNotFound(name, flagsFeature, sdkCore)
+        }
+
+        private fun handleClientNotFound(
+            name: String,
+            flagsFeature: FlagsFeature,
+            sdkCore: FeatureSdkCore
+        ): FlagsClient {
+            val buildHint = if (name == DEFAULT_CLIENT_NAME) {
+                "FlagsClient.Builder().build()"
+            } else {
+                "FlagsClient.Builder(\"$name\").build()"
+            }
+
+            // Client not found, in strict mode this call will crash the app.
+            flagsFeature.logErrorWithPolicy(
+                message = "No FlagsClient with name '$name' exists for SDK instance '${sdkCore.name}'. " +
+                    "Create a client first using: $buildHint. " +
+                    "Operating in no-op mode.",
+                level = InternalLogger.Level.ERROR,
+                shouldCrashInStrict = true
+            )
+
+            // Did not crash, not in strict mode, so return a NoOpFlagsClient that logs through the Feature's policy-driven logger.
+            return NoOpFlagsClient(
+                name = name,
+                reason = "Client '$name' not found - get() called before build()",
+                logWithPolicy = { message, level ->
+                    flagsFeature.logErrorWithPolicy(message, level, shouldCrashInStrict = false)
+                }
+            )
+        }
+
+        // region Internal
+
+        internal const val FLAGS_CLIENT_EXECUTOR_NAME = "flags-client-executor"
+
+        @Suppress("LongMethod")
+        internal fun createInternal(
+            configuration: FlagsConfiguration,
+            featureSdkCore: FeatureSdkCore,
+            flagsFeature: FlagsFeature,
+            name: String
+        ): FlagsClient {
+            val executorService = featureSdkCore.createSingleThreadExecutorService(
+                executorContext = FLAGS_CLIENT_EXECUTOR_NAME
+            )
+
+            val datadogContext = (featureSdkCore as? InternalSdkCore)?.getDatadogContext()
```

Comment:
nit: cast to `InternalSdkCore` can be done without safety control, it is always guaranteed to be implemented.

This will make a need to only validate for null returned `DatadogContext`, also making validation for `clientToken`, `site`, etc. params redundant, because they are all non-`null` actually as per `DatadogContext` type definition.

Reply:
💯 cleans up nicely.